### PR TITLE
Improve message description handling for scratch org cli `not_found` error

### PR DESCRIPTION
**Pull request by lakshmi2506**

### CODE REVIEW

**Comment by lakshmi2506** [comment]
Hi @davidmreed 

Added a descriptive message for the not_found error while creating a scratch org. It is due to not hitting the correct endpoint or doesn't have access to the API version you are using.

Please review the code 
Thank you

File: cumulusci/core/config/scratch_org_config.py [hunk 1]
```diff
@@ -81,13 +81,26 @@ def create_org(self) -> None:
 
         def raise_error() -> NoReturn:
             message = f"{FAILED_TO_CREATE_SCRATCH_ORG}: \n{stdout}\n{stderr}"
+            try:
+                Output = json.loads(stdout)
```
Comment (davidmreed) [inline cumulusci/core/config/scratch_org_config.py:85]: Variable names in Python typically use "`lower_snake_case`".

```suggestion
                output = json.loads(stdout)
```

File: cumulusci/core/config/scratch_org_config.py [hunk 2]
```diff
@@ -81,13 +81,26 @@ def create_org(self) -> None:
...
+                if (
+                    Output.get("message") == "The requested resource does not exist"
+                    and Output.get("name") == "NOT_FOUND"
+                ):
+                    raise ScratchOrgException(
+                        "Check the API Version or endpoint you are interacting"
```
Comment (davidmreed) [inline cumulusci/core/config/scratch_org_config.py:91]:
```suggestion
                        "The Salesforce CLI was unable to create a scratch org. Ensure you are connected using a valid API version on an active Dev Hub."
```

**Review by davidmreed — CHANGES_REQUESTED** [review]
Nice work, @lakshmi2506! Just a couple of requests from me.

**Review by jofsky — APPROVED** [review]

**Pull request merged**